File: 01142/src/main/java/EA4/ZahlBasis7.java
package EA4;
public class ZahlBasis7 {

	private final static int Basis = 7;
	
	public static void main(String[] args) {
		double a=1.0; double b=0.0; double g=0.6789;
		double c; int d;
		
		for (int i=0; i<25; i++) {
			c=(g*a-b)*Basis;
			d=(int) c;
			b=b*Basis+d;
			a=a*Basis;
			System.out.print(d + "   ");
		}
	}
}

File: 01142/src/main/java/ea4/ZahlBasis7.kt
package ea4


private const val BASIS = 7

fun main() {
    var a = 1.0
    var b = 0.0
    val g = 0.6789
    var c: Double
    var d: Int

    repeat(25) {
        c = (g * a - b) * BASIS
        d = c.toInt()
        b = b * BASIS + d
        a *= BASIS
        print("$d   ")
    }
}
